test/e2e/minimatch/minimatch_test.go

```diff
@@ -44,25 +43,17 @@ type testProfile struct {
 	pools []*pb.Pool
 }
 
-func TestMinimatchStartup(t *testing.T) {
+func TestMinimatchNormalMatchGeneration(t *testing.T) {
 	assert := require.New(t)
+	minimatchTc := createMinimatchForTest(t)
```

Comment:
not for this PR - but in future refactors, can minimatch itself have the mmf embeded? at this point, this is no generic minimatch - we are creating this for testing - so effectively, all of that is just test setup.

Reply:
Sure thing. I'm working on improving the test setup along with the test infra but there's still a long way to go :1st_place_medal: